spec/system/page_objects/modals/insert_video.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require "#{Rails.root}/spec/system/page_objects/modals/base"
+
+module PageObjects
+  module Modals
+    class InsertVideo < PageObjects::Modals::Base
+      BODY_SELECTOR = ".insert-video"
+      MODAL_SELECTOR = ".insert-video-modal"
+      def click_add_video_source_button
+        find(".video-sources .simple-list-uploader .btn").click
+      end
+
+      def click_add_poster_button
+        find(".video-poster .simple-list-uploader .btn").click
+      end
+      def click_add_subtitle_button
+        find(".video-subtitles .simple-list-uploader .btn").click
+      end
+
+      def click_primary_button
+        find(".modal-footer .btn-primary").click
```

Comment:
This is a little too generic IMO. The test may start to fail if there are multiple primary buttons in the future.

Reply:
@tgxworld and @tyb-talks this `click_primary_button` code is already in the base modal page object class too, so no need to reproduce here:

https://github.com/discourse/discourse/blob/c197f8c6ee59f56ba9f830388a36ecfd97b7cbd5/spec/system/page_objects/modals/base.rb#L31-L33

I think it's okay for now, but yes we should probably have a more generic `click_footer_button` for the modal base.

As for d-button, I can't find any places where that uses an ID in core either :( So I think the best you can do is adding a unique CSS class, I think that would be best, something like `btn-insert-video__submit`. There should only ever be 1 cancel button so you can rely on Base for that:

https://github.com/discourse/discourse/blob/c197f8c6ee59f56ba9f830388a36ecfd97b7cbd5/spec/system/page_objects/modals/base.rb#L23-L25